src/codegen/statements.rs

```diff
@@ -1162,41 +1166,7 @@ fn try_catch(
                     },
                 );
 
-                cfg.add(
-                    vartab,
-                    Instr::BranchCond {
-                        cond: Expression::Variable {
-                            loc: try_stmt.expr.loc(),
-                            ty: Type::Bool,
-                            var_no: success,
-                        },
-                        true_block: success_block,
-                        false_block: catch_block,
-                    },
-                );
-
-                cfg.set_basic_block(success_block);
-
-                if !try_stmt.returns.is_empty() {
-                    let mut res = Vec::new();
-
-                    for ret in &try_stmt.returns {
-                        res.push(match ret {
-                            (Some(pos), _) => *pos,
-                            (None, param) => vartab.temp_anonymous(&param.ty),
-                        });
-                    }
-
-                    let buf = &Expression::ReturnData { loc: Codegen };
-                    let decoded = abi_decode(&Codegen, buf, &func_returns, ns, vartab, cfg, None);
-                    for instruction in res.iter().zip(decoded).map(|(var, expr)| Instr::Set {
-                        loc: Codegen,
-                        res: *var,
-                        expr,
-                    }) {
-                        cfg.add(vartab, instruction)
-                    }
-                }
+                (polkadot_ret_switch(loc, success, cfg, vartab), func_returns)
```

Comment:
Shouldn't this call be within an if condition? Although Solana does not support try-catches, if, when building for Solana, we ever reach this function, there might be a problem in our code.

Reply:
This is why I added

```rust
    if !ns.target.is_polkadot() {
        unimplemented!()
    }
```

at the top of this function